DOCSP-41448 write update change code (#45)
mongodb/docs-java-rs: source/tutorials/write-update-documents.txt

Update One Document
-------------------

To update a single document in a MongoDB collection, call the ``updateOne()``
method and pass your query filter as the first argument. Then, pass your

> update operators as the second argument to the ``updateOne()`` method. Finally,

Reviewer: S: I think this sentence can be condensed and simplified a bit (Github won't let me apply this suggestion to the whole sentence because it's showing diffs in between). Because we show the example and list the parameters in order, I think we can omit stating the order here. We also show the parameters in order in earlier in the page.

```suggestion
To update a single document in a collection, call the ``updateOne()``
method and pass it your query filter and your
update operators. Then,
```

pass the ``updateOne()`` method to the static ``Mono.from()`` method from the

> ``Mono Publisher``. In {+java-rs+}, you must use ``Publisher`` implementations to control how data
> is transmitted in your application. The ``Mono Publisher`` is a ``Publisher`` implementation from the Project Reactor library. To learn more

Reviewer: S: I think it might be better to swap these two sentences so that the explanation of `Mono` comes right after its first mention. 

```suggestion
``Mono Publisher``. The ``Mono Publisher`` is a ``Publisher`` implementation from the Project Reactor library. In {+java-rs+}, you must use ``Publisher`` implementations to control how data
is transmitted in your application. To learn more
```

about ``Mono``, see `Mono <https://projectreactor.io/docs/core/release/reference/#mono>`__ in the Project
Reactor documentation.

The following example uses ``Updates.set()`` to update the ``name``
value of a matching document from ``"Bagels N Buns"`` to ``"2 Bagels 2 Buns"``:

.. code-block:: java
   :copyable: true

>     Mono.from(restaurants.updateOne(eq("name", "Bagels N Buns"), set("name", "2 Bagels 2 Buns"))).block();

Reviewer: This comment applies to the sentence just above this code examples (Git won't let me comment there):

I think it's a bit confusing to say "The following example uses Updates.set()" when the set call isn't structured in that way. It might be better to say that it's using the updateOne method:

"The following example uses the ``updateOne()`` method to..."

Update Multiple Documents
-------------------------

> To update multiple documents in a MongoDB collection, call the ``updateMany()``
> method and pass your query filter as the first argument. Then, pass your
> update operators as the second argument to the ``updateMany()`` method. Finally,

Reviewer: S: Same as above suggestion

```suggestion
To update multiple documents in a collection, call the ``updateMany()``
method and pass it your query filter and your
update operators. Then,
```

pass the ``updateMany()`` method to the static ``Mono.from`` method from the

> ``Mono Publisher``. In {+java-rs+}, you must use ``Publisher`` implementations to control how data
> is transmitted in your application. The ``Mono Publisher`` is a ``Publisher`` implementation from the Project Reactor library. To learn more

Reviewer: S: Same as above
```suggestion
``Mono Publisher``. The ``Mono Publisher`` is a ``Publisher`` implementation from the Project Reactor library. In {+java-rs+}, you must use ``Publisher`` implementations to control how data
is transmitted in your application. To learn more
```

set to ``true``, the {+driver-short+} inserts a new document if the query filter doesn't 
match any existing documents.

.. code-block:: java
   :copyable: true

    Mono.from(restaurants.updateMany(eq("borough", "Manhattan"), set("borough",

>     "Manhattan (north)"), new UpdateOptions().upsert(true))).block();

Reviewer: Indent the second line of code to make it clear that it's a continutation of the method opened in the first line
